HPXMLtoOpenStudio/resources/airflow.rb

```diff
@@ -1479,155 +1724,128 @@ def self.apply_infiltration_and_ventilation_fans(model, weather, vent_fans_mech,
-    end
-
-    infil_program.addLine('Set Qbath = 0')
-    vent_fans_bath.each do |vent_bath|
-      infil_program.addLine("Set Qbath = Qbath + #{UnitConversions.convert(vent_bath.rated_flow_rate * vent_bath.quantity, 'cfm', 'm^3/s').round(4)} * #{bath_sch_sensors_map[vent_bath.id].name}")
-    end
-
-    infil_program.addLine('Set Qexhaust = Qrange+Qbath')
-    infil_program.addLine('Set Qsupply = 0')
-    infil_program.addLine("Set QWHV_sup = #{UnitConversions.convert(sup_cfm, 'cfm', 'm^3/s').round(4)}")
-    infil_program.addLine("Set QWHV_exh = #{UnitConversions.convert(exh_cfm, 'cfm', 'm^3/s').round(4)}")
-    infil_program.addLine("Set QWHV_bal = #{UnitConversions.convert(bal_cfm, 'cfm', 'm^3/s').round(4)}")
-
-    infil_program.addLine('Set Qexhaust = Qexhaust + QWHV_exh + QWHV_bal + QWHV_ervhrv')
-    infil_program.addLine('Set Qsupply = Qsupply + QWHV_sup + QWHV_bal + QWHV_cfis + QWHV_ervhrv')
-
+    # Calculate Qfan, Qinf_adj
     # Calculate adjusted infiltration based on mechanical ventilation system
-    infil_program.addLine('Set Qfan = (@Max Qexhaust Qsupply)')
-    if Constants.ERIVersions.index(@eri_version) >= Constants.ERIVersions.index('2019')
-      # Follow ASHRAE 62.2-2016, Normative Appendix C equations for time-varying total airflow
-      infil_program.addLine('If Qfan > 0')
-      # Balanced system if the total supply airflow and total exhaust airflow are within 10% of their average.
-      infil_program.addLine('  Set Qavg = ((Qexhaust + Qsupply) / 2.0)')
-      infil_program.addLine('  If ((@Abs (Qexhaust - Qavg)) / Qavg) <= 0.1') # Only need to check Qexhaust, Qsupply will give same result
-      infil_program.addLine('    Set phi = 1')
-      infil_program.addLine('  Else')
-      infil_program.addLine('    Set phi = (Qinf / (Qinf + Qfan))')
-      infil_program.addLine('  EndIf')
-      infil_program.addLine('  Set Qinf_adj = phi * Qinf')
-      infil_program.addLine('Else')
-      infil_program.addLine('  Set Qinf_adj = Qinf')
-      infil_program.addLine('EndIf')
-    else
-      infil_program.addLine('Set Qu = (@Abs (Qexhaust - Qsupply))') # Unbalanced flow
-      infil_program.addLine('Set Qb = Qfan - Qu') # Balanced flow
-      infil_program.addLine('Set Qtot = (((Qu^2) + (Qinf^2)) ^ 0.5) + Qb')
-      infil_program.addLine('Set Qinf_adj = Qtot - Qu - Qb')
+    infil_program = apply_infiltration_adjustment(infil_program, vent_fans_kitchen, vent_fans_bath, sup_cfm_tot, exh_cfm_tot, bal_cfm_tot, erv_hrv_cfm_tot, infil_flow_actuator, range_sch_sensors_map, bath_sch_sensors_map)
+
+    # Address load of Qfan (Qload)
+    # Qload as variable for tracking outdoor air flow rate, excluding recirculation
+    infil_program.addLine('Set Qload = Qfan')
+    (vent_mech_sup_tot + vent_mech_bal_tot + vent_mech_erv_hrv_tot + vent_mech_cfis_tot).each do |vent_mech_supply_side|
+      # Subtract recirculation air flow rate from Qfan, can only come from supply side
```

Comment:
I understand that you're intentionally excluding exhaust, but is there any harm in including it? Won't average_unit_flow_rate minus average_oa_flow_rate simply be zero for an exhaust system?

It would make the code a lot simpler if we could just do `vent_fans_mech.each do |vent_fan_mech| ...`.